runtime/autoload/lsp/request.vim

```diff
@@ -0,0 +1,21 @@
+let g:nvim_lsp_client = -1
+
+function! s:get_client() abort
+   if g:nvim_lsp_client == -1
+      lua << EOF
+local client = require('runtime.lua.lsp.client')
```

Comment:
This dedented block looks ugly, do not use `<<EOF`. Everything which does not fit into one line (`lua require('lsp.client').get_client()`, or, better, just `return luaeval("require('lsp.client').get_client()")` (move `:if` to lua to not split the logic)) should be in a `require`d function.

Also should be `require('lsp.client')`, I do not know why #6789 still is not merged.